### PR TITLE
MCLOUD-5732: Add ECE-Tools warning about MySQL search being used

### PR DESCRIPTION
**Pull request by shiftedreality**
### Description

MySQL search is not recommended for using and will be removed in the next minor version s of Magento. The proper warning should be thrown to highlight this change.

### Fixed Issues (if relevant)

1. https://jira.corp.magento.com/browse/MCLOUD-5732

### Manual testing scenarios

1. Deploy Magento
2. No warning about "MySQL search engine" present
3. Force ECE-Tools to use MySQL search engine (https://devdocs.magento.com/cloud/env/variables-deploy.html#search_configuration)
4. Deploy Magento
5. Warning about "MySQL search engine being used"

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Step/ValidateConfiguration.php [hunk 1]
```diff
@@ -66,9 +67,11 @@ public function execute()
 
     /**
      * Returns all validation messages grouped by validation level.
-     * Converts validation level to integer value using @see Logger::toMonologLevel() method
```
Comment (oshmyheliuk) [inline src/Step/ValidateConfiguration.php:69]: Why did you remove a description?

**Review by oshmyheliuk — DISMISSED** [review]

File: src/Step/ValidateConfiguration.php (hunk 1 above)
Comment (shiftedreality) [reply src/Step/ValidateConfiguration.php:69]: Reverted

**Review by BarnyShergold — DISMISSED** [review]

File: src/Config/Validator/Deploy/DeprecatedSearchEngine.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    /**
+     * @inheritDoc
+     */
+    public function validate(): Validator\ResultInterface
+    {
+        if (SearchEngine::ENGINE_MYSQL === $this->searchEngine->getName()) {
+            return $this->resultFactory->error(
+                '"MySQL" is configured as a search engine. ' . PHP_EOL .
+                'This option is deprecated and will be removed in one of the next versions.'
```
Comment (shiftedreality) [inline src/Config/Validator/Deploy/DeprecatedSearchEngine.php]: @meker12 could you please review this message?

Comment (meker12) [reply src/Config/Validator/Deploy/DeprecatedSearchEngine.php]: Suggested update:

The MySQL search configuration option is deprecated. Use Elasticsearch instead. See the Elasticsearch configuration instructions in the Cloud Guide: https://devdocs.magento.com/cloud/project/project-conf-files_services-elastic.html.

Not sure if you want to include the URL for the instructions.  If we change the URL, we do configure redirect. Maybe it is enough just to mention that location of the information in the Cloud Guide. (I left off Magento to accomodate future branding changes).

Comment (BarnyShergold) [reply src/Config/Validator/Deploy/DeprecatedSearchEngine.php]: Maybe just say - 
The MySQL search configuration option is deprecated. Use Elasticsearch instead. See https://devdocs.magento.com for further information.

I'm quite sure any developer will figure out in 10s what to search for.....

Comment (shiftedreality) [reply src/Config/Validator/Deploy/DeprecatedSearchEngine.php]: Since ECE-Tools can also be used without Magento Cloud (Docker, on-prem, etc) I will remove link completely.

Comment (BarnyShergold) [reply src/Config/Validator/Deploy/DeprecatedSearchEngine.php]: But https://devdocs.magento.com is applicable to all. However, if you want to leave it clean of links to docs then I'm okay with that. A developer should be able to work things out from a simple message

**Review by BarnyShergold — APPROVED** [review]

**Review by oshmyheliuk — APPROVED** [review]

**Comment by oshmyheliuk** [comment]
Please add test scenario

**Comment by mveeramneni** [comment]
QA Approved
